### PR TITLE
Rename telemetry to usage

### PR DESCRIPTION
**Pull request by tsotnet**
Signed-off-by: Tsotne Tabidze <tsotne@tecton.ai>

**What this PR does / why we need it**: This interface only logs anonymized exceptions and usage of the Feast SDK calls. Renaming the module from telemetry -> usage. 

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: sdk/python/feast/usage.py [hunk 1]
```diff
@@ -25,76 +25,72 @@
...
-                        self._telemetry_id = str(uuid.uuid4())
+                        self._usage_id = str(uuid.uuid4())
 
-                        with open(telemetry_filepath, "w") as f:
-                            f.write(self._telemetry_id)
+                        with open(usage_filepath, "w") as f:
+                            f.write(self._usage_id)
                         print(
                             "Feast is an open source project that collects anonymized error reporting and usage statistics. To opt out or learn"
                             " more see https://docs.feast.dev/reference/telemetry"
```
Comment (woop) [inline sdk/python/feast/usage.py:65]: would this link still work?

**Review by achals — APPROVED** [review]
/lgtm

File: sdk/python/feast/usage.py (hunk 1 above)
Comment (tsotnet) [reply sdk/python/feast/usage.py]: oops, forgot to update this after documentation changes

**Review by woop — APPROVED** [review]

**Pull request merged**